logdetective/server/server.py

```diff
@@ -260,3 +273,167 @@ async def analyze_log_stream(build_log: BuildLog):
     stream = await submit_text(PROMPT_TEMPLATE.format(log_summary), stream=True)
 
     return StreamingResponse(stream)
+
+
+@app.post("/webhook/gitlab/job_events")
+async def receive_gitlab_job_event_webhook(
+    job_hook: JobHook, background_tasks: BackgroundTasks
+):
+    """Webhook endpoint for receiving job_events notifications from GitLab
+    https://docs.gitlab.com/user/project/integrations/webhook_events/#job-events
+    lists the full specification for the messages sent for job events."""
+
+    # Handle the message in the background so we can return 200 immediately
+    background_tasks.add_task(process_gitlab_job_event, job_hook)
```

Comment:
still fascinating to me this background tasks functionality is built into a web framework

I hope we'll be able to decode/monitor based on logs what's happening inside

Reply:
Excellent point, I'll create a task for this.